### PR TITLE
rewrite ids under cita chain types

### PR DESCRIPTION
**Pull request by ouwenkg**

### CODE REVIEW

File: cita-chain/types/src/block_tag.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    Hash(H256),
+    Number(BlockNumber),
+    Earliest,
+    Latest,
+    Pending,
+}
+
+impl From<RpcBlockNumber> for BlockTag {
+    fn from(v: RpcBlockNumber) -> BlockTag {
+        match v {
```
Comment (leeyr338) [inline cita-chain/types/src/block_tag.rs]: what means for `v` ?

Comment (ouwenkg) [reply cita-chain/types/src/block_tag.rs]: OK, it should be `n` or `rpc_block_number`.

**Comment by kaikai1024** [comment]
Rename `BlockId` to `BlockTag`?

*I'm not sure it's a good idea. Just worried about if there is license problem in future.*

**Comment by ouwenkg** [comment]
> Rename `BlockId` to `BlockTag` to avoid GPL?
> 
> _I'm not sure it's a good idea. Just worried about if there is license problem in future._

What you worry about exits indeed, any advices about this now?

**Comment by kaikai1024** [comment]
Better than none.

*btw, Use `rename` is better than `rewrite`.*

File: cita-chain/types/src/block_number.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+use cita_types::H256;
+use jsonrpc_types::rpc_types::{BlockNumber as RpcBlockNumber, BlockTag as RpcBlockTag};
+
+pub type TransactionHash = H256;
+pub type BlockNumber = u64;
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum BlockTag {
```
Comment (leeyr338) [inline cita-chain/types/src/block_number.rs]: I feel so strange about this define!
Do you think `Height` and `Hash` is a `Tag`?
If `yes`, it should be included in `Tag` enum.
If `not`, the enum name should not be `BlockTag`.

Comment (leeyr338) [reply cita-chain/types/src/block_number.rs]: I don't have a better idea for now.

Comment (ouwenkg) [reply cita-chain/types/src/block_number.rs]: At present, just let it go. More modified later.

**Pull request merged**